src/test/java/eu/yals/test/TestUtils.java

```diff
@@ -117,16 +104,27 @@ public static void assertEmpty(String message, String string) {
     public static String getTestUrl() {
         final int serverPort = Integer.parseInt(System.getProperty(App.Properties.SERVER_PORT, "8080"));
         final String localUrl;
-        String runMode =
-                System.getProperty(TestApp.Properties.TEST_RUN_MODE, TestApp.RunMode.LOCAL.name());
 
-        if (runMode.equals(TestApp.RunMode.CONTAINER.name())) {
-            localUrl = String.format("http://host.testcontainers.internal:%d", serverPort);
-        } else {
-            localUrl = String.format("http://localhost:%d", serverPort);
+        if(StringUtils.isNotBlank(System.getProperty(TestApp.Properties.TEST_URL, ""))) {
+            return System.getProperty(TestApp.Properties.TEST_URL);
+        }
+
+        TestApp.RunMode runMode = TestApp.RunMode.valueOf(
+                System.getProperty(TestApp.Properties.TEST_RUN_MODE, TestApp.RunMode.LOCAL.name()));
+
+        switch (runMode) {
```

Comment:
Codacy found an issue: [A switch statement does not contain a break](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=6385434)